tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -1399,25 +1404,29 @@ add_segmentby_to_equivalence_class(EquivalenceClass *cur_ec, CompressionInfo *in
 		new_relids = bms_difference(cur_em->em_relids, uncompressed_chunk_relids);
 		new_relids = bms_add_members(new_relids, info->compressed_rel->relids);
 
+#if PG16_LT
 		/*
 		 * And likewise for nullable_relids.  Note this code assumes
 		 * parent and child relids are singletons.
 		 */
-		new_nullable_relids = cur_em->em_nullable_relids;
+		Relids new_nullable_relids = cur_em->em_nullable_relids;
 		if (bms_overlap(new_nullable_relids, uncompressed_chunk_relids))
 		{
 			new_nullable_relids = bms_difference(new_nullable_relids, uncompressed_chunk_relids);
 			new_nullable_relids =
 				bms_add_members(new_nullable_relids, info->compressed_rel->relids);
 		}
+#endif
```

Comment:
Nit: Personally, I would put this in a block just to make sure that `new_nullable_relids` do not "leak out" to the following code.

Reply:
The new_nullable_relids is actually used in a block that follows this. I have now moved this code together with it and both protected by a PG16_LT macro.